File: python/tests/validate_bst/validate_bst_test.py
import unittest

from treenode.treenode import TreeNode
from validate_bst.validate_bst import Solution


class TestSolution(unittest.TestCase):

    def test_IsValid(self):
        solution = Solution()
        root = TreeNode(2)
        root.left = TreeNode(1)
        root.right = TreeNode(3)

        expected = True
        actual = solution.isValidBST(root)
        self.assertEqual(expected, actual)

    def test_IsValidTwo(self):
        solution = Solution()
        root = TreeNode(5)
        root.left = TreeNode(1)
        root.right = TreeNode(4)
        root.right.left = TreeNode(3)
        root.right.right = TreeNode(6)

        expected = False
        actual = solution.isValidBST(root)
        self.assertEqual(expected, actual)

File: go/validateBST/validate_bst_test.go
package main

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestIsValidBST(t *testing.T) {
	root := &TreeNode{Val: 2}
	root.Left = &TreeNode{Val: 1}
	root.Right = &TreeNode{Val: 3}
	actual := isValidBST(root)
	expected := true
	assert.Equal(t, expected, actual)
}

func TestIsValidBST2(t *testing.T) {
	root := &TreeNode{Val: 5}
	root.Left = &TreeNode{Val: 1}
	root.Right = &TreeNode{Val: 4}
	root.Right.Left = &TreeNode{Val: 3}
	root.Right.Right = &TreeNode{Val: 6}

	actual := isValidBST(root)
	expected := false
	assert.Equal(t, expected, actual)
}
